packages/@lwc/babel-plugin-component/src/index.js

```diff
@@ -40,19 +32,43 @@ module.exports = function LwcClassTransform(api) {
             ]);
         },
         visitor: {
-            Program(path, state) {
-                // The LWC babel plugin doesn't play well when associated along with other plugins. We first
-                // run the LWC babel plugin and then run the rest of the transformation.
-                // We ensure the ordering of the visitors by running all traversals from this Program node.
+            ...decorators(api),
```

Comment:
Agreed. I originally wanted to clean this up after this PR is merged, but I already forgot. 
It would be better to add all the visitors explicitly with this PR.